src/processors/default/ipfsMediaUploader.ts

```diff
@@ -0,0 +1,51 @@
+
+import { urlSource } from 'ipfs-http-client';
+import _ from 'lodash'
+
+import { Table } from '../../db/db';
+import { artworkNotOnIpfs, audioNotOnIpfs } from '../../triggers/ipfs';
+import { Clients, Processor } from '../../types/processor';
+import { ProcessedTrack } from '../../types/track';
+
+function processorFunction(sourceField: 'lossyAudioURL' | 'lossyArtworkURL', replaceField: 'lossyAudioIPFSHash' | 'lossyArtworkIPFSHash') {
+  return async (newTriggerItems: any, clients: Clients) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const tracksByFilePath = _.keyBy(newTriggerItems, (track: ProcessedTrack) => new URL(track[sourceField]).pathname.split('/').pop());
+      
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const sources = newTriggerItems.map((track: ProcessedTrack) => {return urlSource(track[sourceField]!)} );
+    const allFiles = await clients.ipfs.client.addAll(sources, {
+      progress: (bytes, path) => console.log(`added ${bytes} bytes of ${path}`),
+      fileImportConcurrency: 10,
+    });
```

Comment:
oh cool the client already handles using URLs as paths and concurrency and nice progress across urls :O?

Reply:
it appears like it would do that but it still seems to upload them sequentially :/ I might have to figure out a way to upload them concurrently with `.add` instead of using `.addAll`

Reply:
that concurrent with .add seems maybe better